dev/src/bulk-writer.ts

```diff
@@ -226,7 +257,23 @@ export class BulkWriter {
    */
   private closed = false;
 
-  constructor(private readonly firestore: Firestore) {}
+  /**
+   * Rate limiter used to throttle requests as per the 500/50/5 rule.
+   */
+  private rateLimiter = new RateLimiter(
+    STARTING_MAXIMUM_OPS_PER_SECOND,
+    RATE_LIMITER_MULTIPLIER,
+    RATE_LIMITER_MULTIPLIER_MILLIS
+  );
+
+  private readonly options: BulkWriterOptions = {};
+
+  constructor(
+    private readonly firestore: Firestore,
+    options?: BulkWriterOptions
```

Comment:
This will likely be an internal constructor, won't it? If so, I prefer if you just spelled out the option here:

```
interface Firestore() {
  bulkWriter(options?: BulkWriterOptions): BulkWriter {
    return new BulkWriter(this, !options?.disableThrottling);
}

class Firestore() {
  // Note that this is no longer the public API, so we can use Denver's suggestion
  constructor(firestore, readonly enableThrottling) {} 
}
```


If you want to go even fancier you could also pass in a no-throttling implementation of RateLimiter. This could be as simple as `new RateLimiter(Math.Infinity)` if I read this correctly.

Reply:
Wow that is clean! Changed to take in `enableThrottling` and use RateLimiter constructor to enable/disable throttling.